examples/weight_compression/openvino/tiny_llama/main.py

```diff
@@ -0,0 +1,211 @@
+import datetime
+import itertools
+from functools import partial
+from pathlib import Path
+from typing import Callable, Iterable, Optional, TypeVar
+
+import numpy as np
+import openvino as ov
+from datasets import load_dataset
+from optimum.intel import OVModelForCausalLM
+from transformers import AutoTokenizer
+from whowhatbench import Evaluator
+
+import nncf
+from nncf.common.logging import nncf_logger
+
+DataItem = TypeVar("DataItem")
+ModelInput = TypeVar("ModelInput")
+
+ROOT = Path(__file__).parent.resolve()
+MODEL_PATH = ROOT / "compressed_model.xml"
+
+COMPRESSION_MODE = nncf.parameters.CompressWeightsMode.INT4_SYM
+MAX_DROP = 0.2
+MIN_GROUP_SIZE = 64
+MAX_GROUP_SIZE = 128
+MIN_RATIO = 0.5
+MAX_RATIO = 1.0
+
+
+def compress_model(ov_model: ov.Model, nncf_dataset: nncf.Dataset, ratio: float, group_size: int) -> ov.Model:
+    """
+    Compress the given OpenVINO model using NNCF weight compression.
+
+    :param ov_model: The original OpenVINO model to be compressed.
+    :param nncf_dataset: A representative dataset for the weight compression algorithm.
+    :param ratio: The ratio between baseline and backup precisions
+    :param group_size: Number of weights (e.g. 128) in the channel dimension
+        that share quantization parameters (scale).
+    :return: The OpenVINO model with compressed weights.
+    """
+    optimized_ov_model = nncf.compress_weights(
+        ov_model.clone(), # we should clone the model because `compress_weights` works in place
+        dataset=nncf_dataset,
+        mode=COMPRESSION_MODE,
+        ratio=ratio,
+        group_size=group_size,
+        sensitivity_metric=nncf.parameters.SensitivityMetric.HESSIAN_INPUT_ACTIVATION,
```

Comment:
In my latest experiments, MAX_ACTIVATION_VARIANCE gives the best results among other metrics.
Potentially, you can search over them too, maybe in next PR.

Reply:
Thanks for the comment! 
For `HESSIAN_INPUT_ACTIVATION` metric, the appropriate `ratio=0.6` and `group_size=64`, elapsed time - 47 minutes.
For `MAX_ACTIVATION_VARIANCE ` metric, the appropriate `ratio=0.9` and `group_size=128`, elapsed time - 30 minutes.